File: ImplementStrStr/ImplementStrStr.cpp
#include <string>

using namespace std;

class Solution {
public:
	int strStr(string haystack, string needle) {
		if (haystack.size() < needle.size())
			return -1;
		if (haystack.size() == 0 || needle.size() == 0)
			return 0;
		int j = 0, i = 0;
		for (; i < haystack.size() - needle.size() + 1; i++) {
			while (j < needle.size()) {
				if (haystack[i + j] == needle[j]) {
					j++;
				}
				else{
					j = 0;
					break;
				}
			}
			if (j == needle.size())
				return i;
		}
		return -1;
	}
};

File: ImplementStrStr/ImplementStrStr.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if len(haystack) < len(needle):
            return -1
        if len(haystack) == 0 or len(needle) == 0:
            return 0
        for i in range(len(haystack)-len(needle)+1):
            j = 0
            while j < len(needle):
                if haystack[i+j] != needle[j]:
                    break
                else:
                    j += 1
            if j == len(needle):
                return i
        return -1